src/VirtoCommerce.WebHooksModule.Data/Services/WebhookService.cs

```diff
@@ -0,0 +1,129 @@
+            {
+                var webHookIds = result.Select(x => x.Id).ToArray();
+                var webHookSuccessCounts = await _feedReader.GetSuccessCountsAsync(webHookIds);
+                var webHookErrorCounts = await _feedReader.GetErrorCountsAsync(webHookIds);
+
+                foreach (var webHook in result)
+                {
+                    webHook.SuccessCount = webHookSuccessCounts.FirstOrDefault(w => w.Key.EqualsInvariant(webHook.Id)).Value;
+                    webHook.ErrorCount = webHookErrorCounts.FirstOrDefault(w => w.Key.EqualsInvariant(webHook.Id)).Value;
+                }
+            }
+
+            return result.ToArray();
+        }
+
+        public async Task SaveChangesAsync(WebHook[] webHooks)
+        {
+            var pkMap = new PrimaryKeyResolvingMap();
+            var changedEntries = new List<GenericChangedEntry<WebHook>>();
+
+            using (var repository = _webHookRepositoryFactory())
+            {
+                var existingIds = webHooks.Where(x => !x.IsTransient()).Select(x => x.Id).ToArray();
+                var originalEntities = await repository.GetWebHooksByIdsAsync(existingIds);
+
+                foreach (var webHook in webHooks)
+                {
+                    var originalEntity = originalEntities.FirstOrDefault(x => x.Id == webHook.Id);
+                    var modifiedEntity = AbstractTypeFactory<WebHookEntity>.TryCreateInstance().FromModel(webHook, pkMap);
+
+                    if (originalEntity != null)
+                    {
+                        changedEntries.Add(new GenericChangedEntry<WebHook>(webHook, originalEntity.ToModel(new WebHook()), EntryState.Modified));
+                        modifiedEntity.Patch(originalEntity);
+
+                        //Force set ModifiedDate property to mark a webHook changed. Special for update when only event list has changes and webHook table hasn't any changes
+                        originalEntity.ModifiedDate = DateTime.UtcNow;
+                    }
+                    else
+                    {
+                        repository.Add(modifiedEntity);
+                        changedEntries.Add(new GenericChangedEntry<WebHook>(webHook, EntryState.Added));
+                    }
+                }
+
+                await repository.UnitOfWork.CommitAsync();
+                pkMap.ResolvePrimaryKeys();
+
+                ClearCache();
+            }
+        }
+
+        public async Task DeleteByIdsAsync(string[] ids)
+        {
+            using (var repository = _webHookRepositoryFactory())
+            {
+                await repository.DeleteWebHooksByIdsAsync(ids);
+                await repository.UnitOfWork.CommitAsync();
+
+                ClearCache();
+            }
+        }
+
+
+        protected virtual void ClearCache()
+        {
+            WebhookCacheRegion.ExpireRegion();
+            WebhookCacheRegion.ExpireRegion();
```

Comment:
Duplicated line. Or should it be specific SearchCache region?

Reply:
Sure, should be SearchCache